src/main/java/backend/resource/TurboLabel.java

```diff
@@ -4,153 +4,185 @@
 
+    public enum Grouping { EXCLUSIVE, NON_EXCLUSIVE, NONE }
+
+    public static final String DEFAULT_COLOR = "ffffff";
     public static final String EXCLUSIVE_DELIMITER = ".";
     public static final String NONEXCLUSIVE_DELIMITER = "-";
 
-    private void ______SERIALIZED_FIELDS______() {
-    }
-
-    private final String actualName;
-    private final String colour;
-
-    private void ______TRANSIENT_FIELDS______() {
-    }
+    private final String fullName;
+    private final String shortName;
+    private final String groupName;
+    private final Grouping grouping;
 
     private final String repoId;
 
-    private void ______CONSTRUCTORS______() {
-    }
+    @SuppressWarnings("PMD")
+    private String colour;
 
     public TurboLabel(String repoId, String name) {
-        this.actualName = name;
-        this.colour = "ffffff";
+        this.fullName = name;
+
+        String[] splitted = splitKeyword(name);
+        this.groupName = splitted[0];
+        this.grouping = determineGrouping(splitted[1]);
+        this.shortName = splitted[2];
+
+        this.colour = DEFAULT_COLOR;
         this.repoId = repoId;
     }
 
     public TurboLabel(String repoId, String colour, String name) {
-        this.actualName = name;
+        this(repoId, name);
         this.colour = colour;
-        this.repoId = repoId;
-    }
-
-    public static TurboLabel nonexclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, false));
-    }
-
-    public static TurboLabel exclusive(String repoId, String group, String name) {
-        return new TurboLabel(repoId, joinWith(group, name, true));
     }
 
     /**
      * Copy constructor
      */
     public TurboLabel(TurboLabel label) {
-        this.actualName = label.getActualName();
+        this(label.getRepoId(), label.getFullName());
         this.colour = label.getColour();
-        this.repoId = label.getRepoId();
     }
 
     public TurboLabel(String repoId, Label label) {
-        this.actualName = label.getName();
+        this(repoId, label.getName());
         this.colour = label.getColor();
-        this.repoId = repoId;
     }
 
     public TurboLabel(String repoId, SerializableLabel label) {
-        this.actualName = label.getActualName();
+        this(repoId, label.getFullName());
         this.colour = label.getColour();
-        this.repoId = repoId;
     }
 
-    private void ______METHODS______() {
+
+    /**
+     * Extracts group name, group delimiter and short name from a keyword
+     * group name is name of the group associated with a keyword
+     * short name is name of the keyword after omitting its group name 
+     * Example: "priority.high" -> group name = "priority", group delimiter = ".", short name = "high"
+     * @param keyword
+     * @return String array of group name, group delimiter and short name
+     */
+    public static String[] splitKeyword(String keyword) {
```

Comment:
This method is not tested directly. Instead the value of `shortName`, `groupName` and `fullName`are tested. I will make the method private since it won't be used anywhere else.